src/main/java/net/oauth/jsontoken/AsyncJsonTokenParser.java

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package net.oauth.jsontoken;
+
+import com.google.common.base.Function;
+import com.google.common.base.Preconditions;
+import com.google.common.util.concurrent.AsyncFunction;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.gson.JsonObject;
+import net.oauth.jsontoken.crypto.Verifier;
+import net.oauth.jsontoken.discovery.AsyncVerifierProviders;
+import java.security.SignatureException;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executor;
+
+/**
+ * The asynchronous counterpart of {@link JsonTokenParser}.
+ * Class that parses and verifies JSON Tokens asynchronously.
+ */
+public class AsyncJsonTokenParser extends AbstractJsonTokenParser {
+  private final AsyncVerifierProviders asyncVerifierProviders;
+  private final Executor executor;
+
+  /**
+   * Creates a new {@link AbstractJsonTokenParser} with a default system clock. The default
+   * system clock tolerates a clock skew of up to {@link SystemClock#DEFAULT_ACCEPTABLE_CLOCK_SKEW}.
+   *
+   * @param asyncVerifierProviders an object that provides signature verifiers asynchronously
+   *   based on a signature algorithm, the signer, and key ids.
+   * @param executor an executor to run the tasks before and after getting the verifiers
+   * @param checker an audience checker that validates the audience in the JSON token.
+   */
+  public AsyncJsonTokenParser(
```

Comment:
This constructor is probably unnecessary. Asking callers to explicitly pass new SystemClock isn't too much of a burden, and removes implicit dependencies on system time.

Reply:
Removed the entire constructor.